### PR TITLE
[ADF-2995] Permissions - Consumer should be able to upload a new version for his file on a private site

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-2995

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by pionnegru** [comment]
@eromano updated

File: lib/content-services/upload/components/upload-button.component.spec.ts [hunk 1]
```diff
@@ -312,4 +312,69 @@ describe('UploadButtonComponent', () => {
             expect(addToQueueSpy).not.toHaveBeenCalled();
         });
     });
+
+    describe('checkPermission()', () => {
+        it('should not check permission when bode id is not set', () => {
```
Comment (eromano) [inline lib/content-services/upload/components/upload-button.component.spec.ts]: there is a misspelling

File: lib/content-services/upload/components/upload-button.component.spec.ts [hunk 2]
```diff
@@ -312,4 +312,69 @@ describe('UploadButtonComponent', () => {
...
+            spyOn(component.error, 'emit');
+
+            component.checkPermission();
+            fixture.detectChanges();
+
+            expect(component.error.emit).toHaveBeenCalledWith('error');
+        });
+    });
+
+    describe('nodeHasPermission()', () => {
```
Comment (eromano) [inline lib/content-services/upload/components/upload-button.component.spec.ts]: I would prefer not refer a method in the describer but at one behaviour

**Comment by pionnegru** [comment]
@eromano updated

**Pull request merged**